tee-worker/core-primitives/stf-executor/src/lib.rs

```diff
@@ -53,27 +53,28 @@ pub mod executor_tests;
 #[cfg(feature = "mocks")]
 pub mod mocks;
 
+pub type RpcResponseValue = Vec<u8>;
+
 /// Execution status of a trusted operation
 ///
 /// In case of success, it includes the operation hash, as well as
 /// any extrinsic callbacks (e.g. unshield extrinsics) that need to be executed on-chain
 ///
 /// Litentry:
 /// we have made a few changes:
-/// - additionally for the success case, we add the encoded rpc response that will be
-///   passed back to the requester when the top status is `InSidechainBlock`
+/// - we add the encoded rpc response that will be passed back to the requester
 /// - for failed top, we apply the parachain effects too
 #[derive(Clone, Debug, PartialEq)]
 pub enum ExecutionStatus {
-	Success(H256, Vec<OpaqueCall>, Vec<u8>),
-	Failure(Vec<OpaqueCall>),
+	Success(H256, Vec<OpaqueCall>, RpcResponseValue),
+	Failure(H256, Vec<OpaqueCall>, RpcResponseValue),
```

Comment:
Two questions:

1. This is now isomorphic to `(H256, Vec<OpaqueCall>, RpcResponseValue, boolean)`; could/should we factor that out? (can be a named enum instead of a naked boolean ofc)
2. Is the success/failure value not included already (albeit encoded) in the `RpcResponseValue`?